MachObfuscator/SymbolMangling/RealWordsMangler/RealWordsExportTrieMangler.swift

```diff
@@ -1,32 +1,60 @@
 import Foundation
 
 protocol RealWordsExportTrieMangling: AnyObject {
-    func mangle(trie: Trie, fillingRootLabelWith labelFill: UInt8) -> Trie
+    func mangle(trie: Trie) -> Trie
 }
 
 final class RealWordsExportTrieMangler: RealWordsExportTrieMangling {
-    private let machOViewDoomEnabled: Bool
+    private let minimumFillValue: UInt8
 
     init(machOViewDoomEnabled: Bool) {
-        self.machOViewDoomEnabled = machOViewDoomEnabled
+        minimumFillValue = machOViewDoomEnabled ? 0 : 1
     }
 
-    func mangle(trie: Trie, fillingRootLabelWith labelFill: UInt8) -> Trie {
-        var trieCopy = trie
+    func mangle(trie: Trie) -> Trie {
+        // In case of Mach-O trie binary-representation, a root trie node doesn't even have a space to store its label.
+        // That's why it is always empty in the `Trie` struct. But it feels unsafe to pass `0` for
+        // `fillingRootLabelWith`, because `0` is a cstring end marker.
+        var mutableTrie = trie
+        mutableTrie.fillRecursively(startingWithFillValue: minimumFillValue,
+                                    minimumFillValue: minimumFillValue)
+        return mutableTrie
+    }
+}
+
+private extension Trie {
+    struct FillResult {
+        var finalFillValue: UInt8
+    }
 
-        trieCopy.label = trieCopy.label.map { _ in
-            labelFill
+    @discardableResult mutating func fillRecursively(startingWithFillValue initialFillValue: UInt8,
+                                                     minimumFillValue: UInt8) -> FillResult {
+        label.fill(with: initialFillValue)
```

Comment:
`label.fill` is a no-op when label is empty, which is possible - there are checks below. Maybe is would be more readable to make this also explicit here?

Reply:
Do you have any suggestion how to do that?

Reply:
simply: `if !label.isEmpty { label = Array(repeating: initialFillValue, count: label.count) }`

Reply:
Or, I think this could reduce readability, by adding unnecessary noise. `label.isEmpty` check is to decide whatever the `finalFillValue` is. I don't feel this check to be good idea. Do you insist? :)

Reply:
no, I do not think that my version is perfect. I thought about splitting entire `fillRecursively` in 2 methods: one for empty label and one for not empty, but the method is too short for it to matter. I think current version is also ok.